File: TumCampusApp/src/de/tum/in/tumcampusapp/models/GroupList.java
package de.tum.in.tumcampusapp.models;

import java.io.Serializable;
import java.util.List;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

/**
 * Wrapper class holding a list of groups of an employee. Note: This model is based on the TUMOnline web service
 * response format for a corresponding request.
 * 
 * @author Vincenz Doelle
 * @review Daniel G. Mayr
 * @review Thomas Behrens
 */
@Root(name = "gruppen")
public class GroupList implements Serializable {

	private static final long serialVersionUID = 2713473533623487005L;

	@ElementList(inline = true, required = false)
	private List<Group> groups;

	public List<Group> getGroups() {
		return groups;
	}

	public void setGroups(List<Group> groups) {
		this.groups = groups;
	}

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/person/model/GroupList.kt
package de.tum.`in`.tumcampusapp.component.tumui.person.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml
import java.io.Serializable

/**
 * Wrapper class holding a list of groups of an employee. Note: This model is
 * based on the TUMOnline web service response format for a corresponding
 * request.
 */

@Xml(name = "gruppen")
data class GroupList(@Element var groups: List<Group>? = null) : Serializable {
    companion object {
        private const val serialVersionUID = 2713473533623487005L
    }
}
